app/guid-file/template.hbs

```diff
@@ -3,25 +3,30 @@
     @isMobile={{this.media.isMobile}}
 >
     <:header>
-        <h3>
-            <OsfLink
+        <div local-class='FlexContainerRow'>
```

Comment:
I think you may be overthinking this a bit. We could just apply `text-overflow: ellipsis; overflow: hidden; white-space: nowrap; max-width: 90vw` to the `h3` element. No need to rearrange the elements.